cla_backend/libs/eligibility_calculator/calculator.py

```diff
@@ -87,14 +86,11 @@ def _do_cfe_civil_check(self):
 
     @staticmethod
     def _is_data_complete_enough_to_call_cfe(case_data):
-        if not hasattr(case_data, "facts"):
-            # facts section is at the root of everything, so we require it call CFE
-            return False
         if EligibilityChecker._under_18_passported(case_data):
             # no more info needed
             return True
-        if not hasattr(case_data.facts, "dependants_young") and not (
-            hasattr(case_data.facts, "on_passported_benefits") and case_data.facts.on_passported_benefits
+        if case_data.facts.dependants_young is None and not (
```

Comment:
In the four "is_complete" methods, we're trying to figure out if the front end has collected enough data. The front end is rather inconsistent - sometimes the key is not there, sometimes it sets it to null, etc. I think that is what we should be checking for.

Whereas this code has been changed to evaluate the field, so the subtle runes we get from the frontend are now filtered through the new layer that applies to defaults to the values. It just so happens that this key is defaulted to None, but if that changed then this line would not work.

So I think these four "is complete" methods should not evaluate the keys - they should continue to use hasattr. The defaulting / key evaluation bevhaviour is for the many other translators.

Reply:
Then `dependants_young` can have default 0, and simplify translate_dependants

Reply:
Really like that idea, I've been struggling with the default values

Reply:
changed to check __dict__ presence - hasattr() is implemented in terms of getattr() not throwing an exception, and getattr() has been overridden